selfdrive/locationd/laikad.py

```diff
@@ -92,20 +91,21 @@ def cache_ephemeris(self, t: GPSTime):
       cloudlog.debug("Cache saved")
       self.last_cached_t = t
 
-  def get_est_pos(self, t, processed_measurements):
-    if self.last_pos_fix_t is None or abs(self.last_pos_fix_t - t) >= 2:
-      min_measurements = 6 if any(p.constellation_id == ConstellationId.GLONASS for p in processed_measurements) else 5
-      pos_fix, pos_fix_residual = calc_pos_fix_gauss_newton(processed_measurements, self.posfix_functions, min_measurements=min_measurements)
-      if len(pos_fix) > 0:
-        self.last_pos_fix_t = t
-        residual_median = np.median(np.abs(pos_fix_residual))
-        if np.median(np.abs(pos_fix_residual)) < POS_FIX_RESIDUAL_THRESHOLD:
-          cloudlog.debug(f"Pos fix is within threshold with median: {residual_median.round()}")
-          self.last_pos_fix = pos_fix[:3]
-          self.last_pos_residual = pos_fix_residual
-        else:
-          cloudlog.debug(f"Pos fix failed with median: {residual_median.round()}. All residuals: {np.round(pos_fix_residual)}")
-    return self.last_pos_fix
+  def get_lsq_fix(self, t, measurements):
+    if self.last_fix_t is None or abs(self.last_fix_t - t) > 0:
+      min_measurements = 6 if any(p.constellation_id == ConstellationId.GLONASS for p in measurements) else 5
+      position_solution, pr_residuals = calc_pos_fix_gauss_newton(measurements, self.posfix_functions, min_measurements=min_measurements)
+      if len(position_solution) < 3:
+        return None
+      position_estimate = position_solution[:3]
+      position_std = np.median(np.abs(pr_residuals)) * np.ones(3)
+      velocity_solution, prr_residuals = calc_vel_fix(measurements, position_estimate, min_measurements=min_measurements)
+      if len(velocity_solution) == 0 or len(prr_residuals) == 0:
```

Comment:
Can you make this check the same as position_solution check? Shouldn't be different?

Reply:
changed it, I will see it in the tests if it has influence